Core/Meta/Definition.cs

```diff
@@ -35,6 +36,33 @@ protected Definition(string name, Span span, string documentation)
         /// The names of types this definition depends on / refers to.
         /// </summary>
         public abstract IEnumerable<string> Dependencies();
+
+        /// <summary>
+        /// Immediate parent of this definition, if it is enclosed in another definition.
+        /// </summary>
+        public Definition? Parent { get; set; }
+
+        /// <summary>
+        /// List of definitions enclosing this definition, from outer to inner. Empty if this is a top level definition.
+        /// </summary>
+        public List<Definition>? Scope
+        {
+            get
+            {
+                if (Parent is null)
+                {
+                    return null;
+                }
```

Comment:
Remove the `?` from the return type, and remove the `if (Parent is null) { return null; }`, and just let this method return an empty list for a top-level definition like the doc-comment says (the while-loop body will run 0 times :+1:).

Reply:
Ah, oversight.